python/ray/ml/examples/tfx_tabular_train_to_serve.py

```diff
@@ -0,0 +1,201 @@
+
+
+imputer1 = SimpleImputer(
+    ["dropoff_census_tract"], strategy="constant", fill_value=17031839100
+)
+imputer2 = SimpleImputer(
+    ["pickup_community_area", "dropoff_community_area"],
+    strategy="constant",
+    fill_value=8,
+)
+imputer3 = SimpleImputer(["payment_type"], strategy="constant", fill_value="Cash")
+imputer4 = SimpleImputer(
+    ["company"], strategy="constant", fill_value="Taxi Affiliation Services"
+)
+imputer5 = SimpleImputer(
+    ["trip_start_timestamp", "trip_miles", "trip_seconds"], strategy="mean"
+)
+
+ohe = OneHotEncoder(
+    columns=[
+        "trip_start_hour",
+        "trip_start_day",
+        "trip_start_month",
+        "dropoff_census_tract",
+        "pickup_community_area",
+        "dropoff_community_area",
+        "payment_type",
+        "company",
+    ],
+    limit={
+        "dropoff_census_tract": 25,
+        "pickup_community_area": 20,
+        "dropoff_community_area": 20,
+        "payment_type": 2,
+        "company": 7,
+    },
+)
+
+
+def fn(df):
+    df["trip_start_year"] = pd.to_datetime(df["trip_start_timestamp"], unit="s").dt.year
+    return df
+
+
+chained_pp = Chain(
+    imputer1,
+    imputer2,
+    imputer3,
+    imputer4,
+    imputer5,
+    ohe,
+    BatchMapper(fn),
+    BatchMapper(lambda x: x.drop(["trip_start_timestamp"], axis=1)),
+)
+
+# This is came up by manually inspecting the dataset after preprocessing.
+# Hopefully we can get rid of it in the future.
+INPUT_SIZE = 120
+# Not sure what to do with partial batch.
+BATCH_SIZE = 1
+EPOCH = 10
+
+
+def build_model():
+    model = tf.keras.models.Sequential()
+    model.add(tf.keras.Input(shape=(INPUT_SIZE,)))
+    model.add(tf.keras.layers.Dense(50, activation="relu"))
+    model.add(tf.keras.layers.Dense(1, activation="sigmoid"))
+    return model
+
+
+def train_loop_per_worker(config):
+    dataset_shard = train.get_dataset_shard("train")
+
+    strategy = tf.distribute.experimental.MultiWorkerMirroredStrategy()
+    with strategy.scope():
+        model = build_model()
+        model.compile(
+            loss="binary_crossentropy",
+            optimizer="adam",
+            metrics=["accuracy"],
+        )
+
+    for epoch in range(EPOCH):
+        tf_dataset = prepare_dataset_shard(
+            dataset_shard.to_tf(
+                label_column="is_big_tip",
+                output_signature=(
+                    tf.TensorSpec(shape=(BATCH_SIZE, INPUT_SIZE), dtype=tf.float32),
+                    tf.TensorSpec(shape=(BATCH_SIZE,), dtype=tf.int64),
+                ),
+                batch_size=BATCH_SIZE,
+            )
+        )
+
+        model.fit(tf_dataset)
+        train.save_checkpoint(epoch=epoch, model=model.get_weights())
+
+
+trainer = TensorflowTrainer(
+    train_loop_per_worker=train_loop_per_worker,
+    scaling_config={"num_workers": 1},
+    datasets={"train": train_dataset},
+    preprocessor=chained_pp,
+)
+result = trainer.fit()
+
+
+async def adapter(request: Request):
```

Comment:
@simon-mo Could you take a look at line 175 - 201 and see if this way of constructing the endpoint and payload is the recommended way?